### PR TITLE
CodeView: add support for imaginary and complex types

### PR DESCRIPTION
**Pull request by rainers**
Maps imaginary types to the floating point types, creates a struct for complex types.

### CODE REVIEW

File: gen/dibuilder.cpp [hunk 1]
```diff
@@ -262,6 +273,46 @@ ldc::DIType ldc::DIBuilder::CreateVectorType(Type *type) {
...
+        break;
+    default:
+        llvm_unreachable(
+            "Unexpected type for debug info in DIBuilder::CreateComplexType");
+    }
+    ldc::DIFile file = CreateFile();
+
+    LLMetadata *elems[] = {
+        CreateMemberType(0, elemtype, file, "re", 0, PROTpublic),
+        CreateMemberType(0, elemtype, file, "im", t->alignsize(), PROTpublic)};
```
Comment (kinke) [inline gen/dibuilder.cpp]: I think there's a chance that `t->alignsize() != getTypeAllocSize(T)` for 80-bit `real`. So I'd suggest using `getTypeAllocSize(T)` here, like below.

Comment (rainers) [inline gen/dibuilder.cpp]: Thanks, changed that. alignsize was even called on the wrong type.
Also removed the failing assert left over from copy 'n paste.

**Comment by kinke** [comment]
Travis and CircleCI retriggered.

**Comment by rainers** [comment]
Thanks, another one checked off the list at https://github.com/ldc-developers/ldc/issues/1716 (but a different one added, too).

**Pull request merged**